asserts/asserts_test.go

```diff
@@ -199,3 +199,61 @@ func (as *assertsSuite) TestSignFormatSanityNonEmptyBody(c *C) {
 	c.Assert(err, IsNil)
 	c.Check(decoded.Body(), DeepEquals, body)
 }
+
+func (as *assertsSuite) TestAssembleRoundtrip(c *C) {
+	encoded := []byte("type: test-only\n" +
+		"authority-id: auth-id2\n" +
+		"primary-key1: key1\n" +
+		"primary-key2: key2\n" +
+		"revision: 5\n" +
+		"header1: value1\n" +
+		"header2: value2\n" +
+		"body-length: 8\n\n" +
+		"THE-BODY" +
+		"\n\n" +
+		"openpgp c2ln")
+	a, err := asserts.Decode(encoded)
+	c.Assert(err, IsNil)
+
+	hs := a.Headers()
+	c.Check(hs, DeepEquals, map[string]string{
+		"type":         "test-only",
+		"authority-id": "auth-id2",
+		"primary-key1": "key1",
+		"primary-key2": "key2",
+		"revision":     "5",
+		"header1":      "value1",
+		"header2":      "value2",
+		"body-length":  "8",
+	})
+
+	// casual later res mutation doesn't trip us
+	delete(hs, "primary-key1")
+	c.Check(a.Header("primary-key1"), Equals, "key1")
```

Comment:
I would make a new test for this bit so the behaviour is super explicit, e.g. something like TestHeadersReturnsCopy.